src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -247,6 +280,324 @@ public void SendAsync_ExpectedDiagnosticCancelledLogging()
 
                 using (DiagnosticListener.AllListeners.Subscribe(diagnosticListenerObserver))
                 {
+                    diagnosticListenerObserver.Enable(s => !s.Contains("Activity"));
+                    using (var client = new HttpClient())
+                    {
+                        LoopbackServer.CreateServerAsync(async (server, url) =>
+                        {
+                            CancellationTokenSource tcs = new CancellationTokenSource();
+                            Task request = LoopbackServer.AcceptSocketAsync(server,
+                                (s, stream, reader, writer) =>
+                                {
+                                    tcs.Cancel();
+                                    return LoopbackServer.ReadWriteAcceptedAsync(s, reader, writer);
+                                });
+                            Task response = client.GetAsync(url, tcs.Token);
+                            await Assert.ThrowsAnyAsync<Exception>(() => Task.WhenAll(response, request));
+                        }).Wait();
+                    }
+                }
+                // Poll with a timeout since logging response is not synchronized with returning a response.
+                WaitForTrue(() => cancelLogged, TimeSpan.FromSeconds(1),
+                    "Cancellation was not logged within 1 second timeout.");
```

Comment:
This timeout is used in other tests here, which were running for quite a while and seems to be stable, at lease I cannot find any issues for test failures
Note that response is awaited, and response events are fired after response is received, so I'm not even convinced here should be any timeout here.